### PR TITLE
Rad 1237

### PR DESCRIPTION
**Pull request by MarcusEHughes**
Added support for template.type parameter

### CODE REVIEW

**Review by LouwMarais — APPROVED** [review]

**Comment by maraisr** [comment]
Looks good from me 💯

**Comment by MarcusEHughes** [comment]
It's Marais! :)

File: .gitignore [hunk 1]
```diff
@@ -26,4 +26,5 @@ target/
 build/
 bin/
 classes/
-generated/
\ No newline at end of file
+generated/
```
Comment (srowatt) [inline .gitignore]: remove generated and out dirs

Comment (MarcusEHughes) [reply .gitignore]: As you would say so eloquently, nahhhh

**Pull request merged**